File: app/src/main/java/com/esbati/keivan/persiancalendar/Models/CalendarDay.java
package com.esbati.keivan.persiancalendar.Models;

import java.util.ArrayList;

import ir.smartlab.persindatepicker.util.PersianCalendar;

/**
 * Created by asus on 11/20/2016.
 */

public class CalendarDay {
    public int mDayNo;
    public boolean isToday;
    public boolean isHoliday;
    public boolean isCurrentMonth;
    public PersianCalendar mPersianDate;
    public ArrayList<CalendarEvent> mCalendarEvents;
    public ArrayList<GoogleEvent> mGoogleEvents;

    public CalendarDay(PersianCalendar persianCalendar){
        mPersianDate = persianCalendar;

        if(mPersianDate != null)
            mDayNo = mPersianDate.getPersianDay();
    }

    public CalendarDay(int day){
        mDayNo = day;
    }
}

File: app/src/main/java/com/esbati/keivan/persiancalendar/pojos/CalendarDay.kt
package com.esbati.keivan.persiancalendar.pojos

import java.util.*

/**
 * Created by Keivan Esbati on 11/20/2016.
 */

class CalendarDay(
    val mYear: Int,
    val mMonth: Int,
    val mDay: Int,
    val isToday: Boolean = false,
    val isHoliday: Boolean = false,
    val isCurrentMonth: Boolean = false,
    val formattedDate: String = "",
    val formattedDateSecondary: String = "",
    val mRemarks: ArrayList<CalendarRemark> = ArrayList(),
    val mEvents: ArrayList<UserEvent> = ArrayList())